### PR TITLE
added some missing ctor initializers

### PR DESCRIPTION
**Pull request by esohns**

### CODE REVIEW

**Comment by esohns** [comment]
please mind only commit 90cf05a. Something went wrong during branch creation.

File: ACE/ace/Timer_Heap_T.cpp [hunk 1]
```diff
@@ -99,6 +99,7 @@ ACE_Timer_Heap_T<TYPE, FUNCTOR, ACE_LOCK, TIME_POLICY>::ACE_Timer_Heap_T (
     max_size_ (size),
     cur_size_ (0),
     cur_limbo_ (0),
+    iterator_ (NULL),
```
Comment (jwillemsen) [inline ACE/ace/Timer_Heap_T.cpp:102]: NULL is not portable, given that we mandate C++11 please use nullptr

File: ACE/ace/Message_Block.cpp [hunk 2]
```diff
@@ -1204,33 +1204,27 @@ ACE_Message_Block::clone (Message_Flags mask) const
...
-                                             ACE_DEADLINE_TIME, // absolute time to deadline
-                                             // Get a pointer to a
-                                             // "duplicated"<ACE_Data_Block>
-                                             // (will simply increment the
-                                             // reference count).
-                                             db,
-                                             db->data_block_allocator (),
-                                             old_message_block->message_block_allocator_),
-                          0);
+          ACE_NEW_NORETURN (new_message_block,
```
Comment (jwillemsen) [inline ACE/ace/Message_Block.cpp:1207]: Please remove this from this PR, this is already part of another one

File: ACE/ace/Timer_Heap_T.cpp (hunk 1 above)
Comment (jwillemsen) [reply ACE/ace/Timer_Heap_T.cpp:102]: The fuzz build gives an error about this

**Review by jwillemsen — CHANGES_REQUESTED** [review]
See comments

**Pull request merged**